walkthroughs/howto-k8s-tls-acm/README.md

```diff
@@ -0,0 +1,224 @@
+    --signing-algorithm SHA256WITHRSA \
+    --validity Value=10,Type=YEARS \
+    --csr "${ROOT_CA_CSR}" \
+    --query CertificateArn --output text`
+```
+
+Then import the signed certificate as the root CA:
+
+```bash
+ROOT_CA_CERT=`aws acm-pca get-certificate \
+    --certificate-arn ${ROOT_CA_CERT_ARN} \
+    --certificate-authority-arn ${ROOT_CA_ARN} \
+    --query Certificate --output text`
+```
+
+Note again with AWS CLI version 2, you will need to pass the certificate data through encoding.
+
+```bash
+[[ ${AWS_CLI_VERSION} -gt 1 ]] && ROOT_CA_CERT="$(echo ${ROOT_CA_CERT} | base64)"
+```
+
+Import the certificate:
+
+```bash
+aws acm-pca import-certificate-authority-certificate \
+    --certificate-authority-arn $ROOT_CA_ARN \
+    --certificate "${ROOT_CA_CERT}"
+```
+
+We also want to grant permissions to the CA to automatically renew the managed certificates it issues:
+
+```bash
+aws acm-pca create-permission \
+    --certificate-authority-arn $ROOT_CA_ARN \
+    --actions IssueCertificate GetCertificate ListPermissions \
+    --principal acm.amazonaws.com
+```
+
+Request a managed certificate "color-blue.howto-k8s-tls-file-based.svc.cluster.local" from ACM using this CA:
+
+```bash
+export BLUE_CERTIFICATE_ARN=`aws acm request-certificate \
+    --domain-name "color-blue.${SERVICES_DOMAIN}" \
+    --certificate-authority-arn ${ROOT_CA_ARN} \
+    --query CertificateArn --output text`
+```
+
+Request a managed certificate "color-green.howto-k8s-tls-file-based.svc.cluster.local" from ACM using this CA
+```bash
+export GREEN_CERTIFICATE_ARN=`aws acm request-certificate \
+    --domain-name "color-green.${SERVICES_DOMAIN}" \
+    --certificate-authority-arn ${ROOT_CA_ARN} \
+    --query CertificateArn --output text`
+```
+
+With managed certificates, ACM will automatically renew certificates that are nearing the end of their validity, and App Mesh will automatically distribute the renewed certificates. See [Managed Renewal and Deployment](https://aws.amazon.com/certificate-manager/faqs/#Managed_Renewal_and_Deployment) for more information.
+
+## Step 3: Create a Mesh with TLS enabled
+
+We are going to setup a mesh with four virtual nodes: Frontend, Blue, Green and Red, one virtual service: color and one virtual router: color.
+
+Let's create the mesh.
+
+```bash
+./mesh.sh up
+```
+
+Frontend has backend virtual service (color) configured and the virtual service (color) uses virtual router (color) as the provider. The virtual router (color) has three routes configured:
+- color-route-blue: matches on HTTP header "blue" to route traffic to virtual node `blue`
+- color-route-green: matches on HTTP header "green" to route traffic to virtual node `green` 
+- color-route-red: matches on HTTP header "red" to route traffic to virtual node `red`
+
+Virtual nodes `blue` and `green` are configured with TLS enabled for their respective listeners. Here's the spec for `green` Virtual Node:
```

Comment:
Is there a reason to only configure this for 2 of these? It's just a demo, so good to see that it is a per-node configuration parameter, so I'm not passionate about configuring all services. But maybe worth a note? Others might have stronger opinions (or not).

Reply:
Agreed. From demo perspective, displaying a single virtual node integration should be enough. The doc and the example has been updated to use a wildcard certificate and only one of the virtual node is updated with the TLS config

Reply:
I had actually meant "why not all services", but one is fine as well.